lib/mapObjects/MiscObjects.cpp

```diff
@@ -1324,6 +1327,11 @@ void CGArtifact::onHeroVisit(const CGHeroInstance * h) const
 				}
 			}
 			break;
+			}
+		}
+		else
+		{
+			iw.text << "no slots";
```

Comment:
```suggestion
			iw.text.addTxt(MetaString::ADVOB_TXT,2);
```

Also, not sure whether H3 removes artifact from map if you have full backpack or keeps it.

Reply:
@IvanSavenko I would suggest keep, even it it is a deviation. TheLazy indirectly says than artefact remains: https://heroes.thelazy.net/index.php/Artifact

Reply:
Not sure. In theory this may break the game, e.g. if this artifact is needed for quest. Silently removing is not a good idea either - player might want to come back later for relic if his backpack is full of garbage arts.

What we *can* do in theory:
- query player what to do with artifact - discard or keep on ground
- allow deletion of artifacts in backpack, so player can destroy artifact and pick artifact from ground.

Reply:
In H3 artifacts are not removed from the map. Like in this PR.